Automattic/wp-calypso
# Statsd: record multiple events at once

### Proposed Changes
This PR accomplishes a few things:

1. Refactor the statsd helpers to accept multiple events at once.
2. Update the place we use those helpers to use the new format.
3. Remove the server-side statsd wrapper because it wasn't adding value.

### Statsd helper refactor

If you search across the repo for `statsdCountingUrl` or `statsdTimingUrl`, they are only used by the server-side analytics. Then, the server-side analytics `recordTiming` or `recordCounting` are only used by the post-request logging function.

As a result, this refactor is very low impact. But my hope is that it makes it easier to use statsd in the future because it's more consistent. An event now has this very basic structure:

```ts
interface BeaconData {
	name: string;
	value: number;
	type: 'counting' | 'timing';
}
```

Previously, a function signature was `function recordCounting( featureSlug, eventType, duration )`. It's unclear what data should be in the "featureSlug" or what an "eventType" is. How do I add data which can be filtered in grafana? Is that part of the "feature?" Do I have to specify a known "eventType"? The Calypso section is the same as the "featureSlug" -- should I add that? 

The new approach exposes things a little more clearly -- you always log the section name, and can log multiple events, each of which has a name, value, and type. The name includes everything you'd use to filter the event. You also always import from `calypso/lib/analytics/statsd-utils` instead of a custom server analytics lib.

IMO, this is easier to use without looking up documentation. Plus, this approach makes it very straightforward to log multiple events without extra network requests, and I figured it made more sense to have it all in one function rather than have multiple functions for single events and multiple events and different types of events.

### Extra event data
_I will add this to a follow-up PR to simplify this one._

~I'd like to get extra data into our Calypso server response time dashboard. (Search for "calypso node response times" in Grafana to find it.) In particular, I'd like to visualize response time of SSR sections vs non-SSR sections and logged-in vs logged-out requests.~

~Periods in an event name are used to filter things. So we just need an event like this: `calypso.production.${section}.loggedin_${true|false}.ssr_${true|false}.response_time:${value}`~

~Since this isn't backwards compatible with the existing dashboard queries (`calypso.production.*.response_time`), we add it as an extra event. (Which is now very easy to do thanks to the above refactor!)~

#### Testing Instructions
1. Test calypso.live and verify various responses still work.
2. Verify that your requests to calypso.live show up in Calypso server analytics. (TBD)
